docs: Added documentation of Expression Monitors (#178)
scrapinghub/spidermon: docs/source/settings.rst

SPIDERMON_EXPRESSIONS_MONITOR_CLASS
-----------------------------------

> 

Reviewer: :lipstick: A single empty line should be enough.

Default: ``spidermon.python.monitors.ExpressionMonitor``

> A fully qualified path for a Python Class that should inherit from 

Reviewer: Instead of ‘Python Class that should inherit from’, you could just say ‘subclass of’.

``spidermon.python.monitors.ExpressionMonitor``.

> This class will be used to generate on-the-fly monitors for the folowing settings:

Reviewer: I would just go with:

```suggestion
This class will be used to generate :ref:`expression monitors <expression-monitors>`.
```

And remove the remaining text below (list and paragraph after the list). Users interested in expression monitors can follow the link and find everything they need.

    * :ref:`SPIDERMON_SPIDER_CLOSE_EXPRESSION_MONITORS`
    * :ref:`SPIDERMON_SPIDER_OPEN_EXPRESSION_MONITORS`
    * :ref:`SPIDERMON_ENGINE_STOP_EXPRESSION_MONITORS`

Please refer to :ref:`expression-monitors` to better understand what is
a Expression Monitor.

.. note::
    Your probably wont change this setting unless you have an advanced use case and
    needs to change how the context data is build or how the on-the-fly ``MonitorSuite``

>     are generated. Otherwise the default should be enough

Reviewer: I think this note is really helpful, although in the future, when we extend the documentation about expression monitors to cover when and how to change this setting, I would remove this note as well.

SPIDERMON_SPIDER_CLOSE_MONITORS
-------------------------------

Default: ``[]``

List of monitor suites to be executed when the spider closes.

.. _SPIDERMON_SPIDER_CLOSE_EXPRESSION_MONITORS:

SPIDERMON_SPIDER_CLOSE_EXPRESSION_MONITORS
------------------------------------------

Default: ``[]``

> List of dictionaries, each dictionary stands for a MonitorSuite with one

Reviewer: What about a single paragraph like this?

```
List of dictionaries describing :ref:`expression monitors <expression-monitors>` to run when a spider is closed.
```

A similar approach could be followed for the other related settings.
